Source/Core/VideoCommon/Fifo.cpp

```diff
@@ -386,11 +420,20 @@ bool AtBreakpoint()
 
 void RunGpu()
 {
+	SCPFifoStruct &fifo = CommandProcessor::fifo;
 	if (SConfig::GetInstance().m_LocalCoreStartupParameter.bCPUThread &&
 	    !g_use_deterministic_gpu_thread)
+	{
+		if (!s_gpu_is_running.load())
+		{
+			std::unique_lock<std::mutex> lock(s_fifo_mutex);
```

Comment:
I think there is a race condition here.

If this lock is somehow acquired just before line 380, but after line 374 then we will hold this lock preventing the condition variable from waiting, notify nothing and then the gpu thread will continue and deadlock on waiting for notify, which the cpu thread will never generate because s_gpu_is_running is true.